File: LeetCodePython/88.合并两个有序数组.py
#
# @lc app=leetcode.cn id=88 lang=python3
#
# [88] 合并两个有序数组
#
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        pos = m + n - 1 
        while n>0:
            if m > 0 and nums1[m-1] > nums2[n-1]:
                nums1[pos] = nums1[m-1]
                m -= 1
            else:
                nums1[pos] = nums2[n-1]
                n -= 1
            pos -= 1
            

        

File: LeetCodeCPlusPlus/88.合并两个有序数组.cpp
/*
 * @lc app=leetcode.cn id=88 lang=cpp
 *
 * [88] 合并两个有序数组
 *
 * https://leetcode-cn.com/problems/merge-sorted-array/description/
 *
 * algorithms
 * Easy (45.03%)
 * Likes:    293
 * Dislikes: 0
 * Total Accepted:    62.2K
 * Total Submissions: 138.2K
 * Testcase Example:  '[1,2,3,0,0,0]\n3\n[2,5,6]\n3'
 *
 * 给定两个有序整数数组 nums1 和 nums2，将 nums2 合并到 nums1 中，使得 num1 成为一个有序数组。
 * 
 * 说明:
 * 
 * 
 * 初始化 nums1 和 nums2 的元素数量分别为 m 和 n。
 * 你可以假设 nums1 有足够的空间（空间大小大于或等于 m + n）来保存 nums2 中的元素。
 * 
 * 
 * 示例:
 * 
 * 输入:
 * nums1 = [1,2,3,0,0,0], m = 3
 * nums2 = [2,5,6],       n = 3
 * 
 * 输出: [1,2,2,3,5,6]
 * 
 */
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int pos = m + n - 1;
        while (n>0) {
            if (m>0 && nums1[m-1] > nums2[n-1]) {
                nums1[pos] = nums1[m-1];
                m--;
            } else {
                nums1[pos] = nums2[n-1];
                n--;
            }
            pos--;
        }
    }
};

